bcipy/gui/mode/RSVPKeyboard.py

```diff
@@ -1,210 +1,281 @@
 # pylint: disable=no-member
 """GUI for running RSVP tasks"""
-import datetime
 import itertools
 import os
 import subprocess
+import sys
 
-import wx
+from typing import List
 
-from bcipy.gui.gui_main import BCIGui
+from bcipy.gui.gui_main import BCIGui, app, AlertMessageType, AlertResponse
 from bcipy.helpers.load import load_json_parameters, copy_parameters
 from bcipy.helpers.parameters import DEFAULT_PARAMETERS_PATH
-
 from bcipy.tasks.task_registry import ExperimentType
 
+
 class RSVPKeyboard(BCIGui):
-    """GUI for launching the RSVP tasks."""
+    """GUI for launching the RSVP tasks and editing parameters."""
+
+    tasks = ExperimentType.by_mode()['RSVP']
+
+    default_text = '...'
+    padding = 15
+    btn_width_apart = 105
+    btn_padding = 20
 
     def __init__(self, *args, **kwargs):
         super(RSVPKeyboard, self).__init__(*args, **kwargs)
         self.event_started = False
         self.parameter_location = DEFAULT_PARAMETERS_PATH
 
-    def bind_action(self, action: str, btn) -> None:
-        if action == 'launch_bci':
-            self.Bind(wx.EVT_BUTTON, self.launch_bci_main, btn)
-        elif action == 'edit_parameters':
-            self.Bind(wx.EVT_BUTTON, self.edit_parameters, btn)
-        elif action == 'select_parameters':
-            self.Bind(wx.EVT_BUTTON, self.select_parameters, btn)
-        elif action == 'refresh':
-            self.Bind(wx.EVT_BUTTON, self.refresh, btn)
-        elif action == 'offline_analysis':
-            self.Bind(wx.EVT_BUTTON, self.offline_analysis, btn)
-        elif action == 'load_items_from_txt':
-            self.Bind(wx.EVT_COMBOBOX_DROPDOWN, self.load_items_from_txt, btn)
-        else:
-            self.Bind(wx.EVT_BUTTON, self.on_clicked, btn)
-
-    def select_parameters(self, _event) -> None:
-        """Dialog to select the parameters.json configuration to use."""
-        with wx.FileDialog(self,
-                           'Select parameters file',
-                           wildcard='JSON files (*.json)|*.json',
-                           style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST) as fd:
-            if fd.ShowModal() != wx.ID_CANCEL:
-                self.parameter_location = fd.GetPath()
-
-    def edit_parameters(self, _event) -> None:
-        """Edit Parameters. Prompts for a parameters.json file to use. If the default parameters
-        are selected a copy is used.
+        self.parameters = load_json_parameters(
+            self.parameter_location, value_cast=True)
+
+        self.task_colors = itertools.cycle(
+            ['blue', 'green', 'yellow', 'red', 'limegreen', 'gray', 'pink'])
```

Comment:
These colors seem more intense in pyqt than wx. We may need a better palette.

Reply:
I quickly added some to test the refactor. I'll look for something a little more appealing!